tests/test_pipelines.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python
+
+#
+# LSST Data Management System
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+"""Test cases for cp_pipe pipelines."""
+
+import os
+import unittest
+
+from lsst.pipe.base import Pipeline, PipelineGraph
+import lsst.utils
+
+try:
+    import lsst.obs.lsst
+    has_obs_lsst = True
+except ImportError:
+    has_obs_lsst = False
+
+try:
+    import lsst.obs.subaru
+    has_obs_subaru = True
+except ImportError:
+    has_obs_subaru = False
+
+try:
+    import lsst.obs.decam
+    has_obs_decam = True
+except ImportError:
+    has_obs_decam = False
+
+
+class CalibrationPipelinesTestCase(lsst.utils.tests.TestCase):
+    """Test case for building the pipelines."""
+
+    def setUp(self):
+        self.pipeline_path = os.path.join(lsst.utils.getPackageDir("cp_pipe"), "pipelines")
+
+    def _get_pipelines(self, exclude=[]):
```

Comment:
I think this either needs to be able to auto-discover pipelines (possibly with a `glob(os.path.join(self.pipeline_path, "*", "*.yaml"))`, or we need a note in the `README.md` that newly added pipelines need to be added to this list.
Hopefully we won't have many more to add, but we don't want things to get out of sync.